### PR TITLE
Allow overriding paths to executables using env vars.

### PR DESCRIPTION
**Pull request by maleadt**
I'm working on distributing pocl for use with the Julia programming language. One of the issues I ran into, is that pocl embeds hard-coded paths to binaries like `llvm-spirv` and `clang` in the binary with values set at configuration time. That doesn't work for our use case, where the compile-time paths in our build container doesn't match the run-time situation (which may differ from user to user, as Julia installs binary artifacts in `~/.julia/artifacts`).

In this PR, I propose adding the ability to override paths to executables through `POCL_XXX` environment variables. At initialization time, our pocl wrapper package populates those variables with paths to wrapper scripts that point to our binaries (in addition with some set-up for library paths).

### CODE REVIEW

File: lib/CL/clCreateProgramWithIL.c [hunk 1]
```diff
@@ -37,7 +37,8 @@
 static int
 get_program_spec_constants (cl_program program, char *program_bc_spirv)
 {
-  char *args[] = { LLVM_SPIRV, "--spec-const-info", program_bc_spirv, NULL };
+  char *args[] = {get_llvm_spirv(), "--spec-const-info", program_bc_spirv,
```
Comment (pjaaskel) [inline lib/CL/clCreateProgramWithIL.c:40]: Did you run clang-format (format-diff/branch.sh)? Not sure if it prefers to have the white space in the beginning/end of the list.

**Review by pjaaskel — COMMENTED** [review]
In principle this is OK, but I started to think whether having separate envs for each binary is a standard practice or should the user adjust their PATH to point / prioritize the direction of the binaries and then we'd implement a possibility to define relative paths to the executable in PoCL?

File: lib/CL/clCreateProgramWithIL.c (hunk 1 above)
Comment (maleadt) [reply lib/CL/clCreateProgramWithIL.c:40]: I ran `git clang-format`, assuming that there would be a top-level `.clang-format`. Apparently the workflow here is different, I'll re-format the changes.

Comment (pjaaskel) [reply lib/CL/clCreateProgramWithIL.c]: Yep, unfortunately we have different styles for C and C++.

**Comment by maleadt** [comment]
> I started to think whether having separate envs for each binary is a standard practice or should the user adjust their PATH to point / prioritize the direction of the binaries and then we'd implement a possibility to define relative paths to the executable in PoCL

I'd rather not pollute the user's PATH with entries that may end up affecting other code. Lots of env vars are kinda ugly, but at least they allow minimizing the impact. Otherwise I'd have to set PATH for the duration of the entire application, or across every OpenCL API call.

Alternatively, a `POCL_PREFIX` or `POCL_PATH` variable with a list of candidate directories to look for executables in could be another approach, but that would complicate the change.

**Comment by maleadt** [comment]
Thanks for the suggestions. Inspired by `pocl_get_string_option` I moved to a similar `pocl_get_path` defined in `pocl_runtime_config.c`. That seems much less invasive of a change.

Since `pocl_get_path` should return a const string, I had to change some other signatures for const correctness, necessitating a questionable cast around the call to `execv` (which seems correct though, https://stackoverflow.com/questions/190184/execv-and-const-ness).

Also added docs. I'll detail the Julia situation in a future PR when we have everything figured out, if you don't mind.

**Comment by maleadt** [comment]
Didn't have LEVEL0 enabled locally, so missed a couple of const correctness issues (for some reason the relevant `-Werror` doesn't enable here locally, so its possible I missed some more).

Can you re-trigger CI?

**Comment by maleadt** [comment]
Same think for the Vulkan back-end. I think I got them all now.

**Comment by maleadt** [comment]
Remaining failures seem unrelated.

**Pull request merged**